src/operations/command.ts

```diff
@@ -141,6 +139,12 @@ export abstract class CommandOperation<
       this.logger.debug(`executing command ${JSON.stringify(cmd)} against ${this.ns}`);
     }
 
+    // If a command is to be explained, we need to reformat the command after
+    // the other command properties are specified.
+    if (cmd.explain) {
+      cmd = decorateWithExplain(cmd, cmd.explain);
+    }
+
     server.command(
       this.ns.toString(),
       cmd,
```

Comment:
This is a stylistic nit, but I think it'd be better to do `cmd.explain ? decorateWithExplain(cmd, cmd.explain) : cmd` here rather than reassigning the `cmd` parameter.